libvips/colour/icc_transform.c

```diff
@@ -840,44 +839,50 @@ vips_icc_import_build( VipsObject *object )
 	return( 0 );
 }
 
-static void 
+static void
 decode_lab( guint16 *fixed, float *lab, int n )
 {
 	int i;
 
-        for( i = 0; i < n; i++ ) {
-                lab[0] = (double) fixed[0] / 652.800;
-                lab[1] = ((double) fixed[1] / 256.0) - 128.0;
-                lab[2] = ((double) fixed[2] / 256.0) - 128.0;
+	for( i = 0; i < n; i++ ) {
+		cmsCIELab Lab;
+		cmsLabEncoded2Float( &Lab, fixed );
 
-                lab += 3;
-                fixed += 3;
-        }
+		lab[0] = (float) Lab.L;
+		lab[1] = (float) Lab.a;
+		lab[2] = (float) Lab.b;
+
+		lab += 3;
+		fixed += 3;
+	}
 }
 
-#define X_FAC (VIPS_D50_X0 * 32768 / (VIPS_D65_X0 * 100))
-#define Y_FAC (VIPS_D50_Y0 * 32768 / (VIPS_D65_Y0 * 100))
-#define Z_FAC (VIPS_D50_Z0 * 32768 / (VIPS_D65_Z0 * 100))
+#define X_FAC (cmsD50X / VIPS_D65_X0)
```

Comment:
It looks like these constants are always used in a float context so prefixing with a `(float)` cast would prevent an implicit conversion to double, although I'm unsure if this would help performance.